### PR TITLE
BFS optimization and memory access count script update

### PR DESCRIPTION
**Pull request by ywwu928**
# Description

## PR description

This PR _write a high-level description of your pull request_

## Checklist

- [v] The additions follow the code standards in the [developer guide](pando-rt/docs/developer.md).
- [v] New or existing tests cover these changes.
- [v] The documentation is up to date with these changes.

### CODE REVIEW

File: microbench/bfs/include/pando-bfs-galois/sssp.hpp [hunk 1]
```diff
@@ -285,11 +290,20 @@ pando::Status SSSPMDLCSR(G& graph, std::uint64_t src, HostLocalStorage<MDWorkLis
         auto [srcID, found] = graph.getLocalTopologyID(src);
         if (found) {
           graph.setDataOnly(srcID, 0);
-          PANDO_CHECK(fmap(toRead[0], pushBack, srcID));
+
+          std::int16_t srcHost = graph.getPhysicalHostID(src);
+          if (srcHost == pando::getCurrentPlace().node.id) {
```
Comment (AdityaAtulTewari) [inline microbench/bfs/include/pando-bfs-galois/sssp.hpp:295]: this is wrong.

File: microbench/bfs/src/main.cpp [hunk 2]
```diff
@@ -104,6 +104,15 @@ void HBMainMDLCSR(pando::Vector<std::uint64_t> srcVertices, std::uint64_t numVer
   PANDO_CHECK(toRead.initialize());
   PANDO_CHECK(toWrite.initialize());
 
+#ifdef DPRINTS
+  // print out number of mirrors
+  PANDO_CHECK(galois::doAll(
+      graph, toRead, +[](Graph graph, bfs::MDWorkList<Graph>) {
+        std::cout << "Host " << pando::getCurrentPlace().node.id << " has " << graph.getMirrorSize()
+                  << " mirrors" << std::endl;
```
Comment (AdityaAtulTewari) [inline microbench/bfs/src/main.cpp:112]: std::cerr or it breaks correctness checks.

File: microbench/bfs/include/pando-bfs-galois/sssp.hpp (hunk 1 above)
Comment (ywwu928) [reply microbench/bfs/include/pando-bfs-galois/sssp.hpp:295]: Why? If I don't add that line, one of the test will break.
You should only add the master of the source vertex to the worklist but not the mirror in OEC.

File: microbench/bfs/include/pando-bfs-galois/sssp.hpp [hunk 3]
```diff
@@ -285,11 +290,20 @@ pando::Status SSSPMDLCSR(G& graph, std::uint64_t src, HostLocalStorage<MDWorkLis
...
+            PANDO_CHECK(fmap(toRead[0], pushBack, srcID));
+          }
         }
       }));
   PANDO_CHECK_RETURN(wg.wait());
 
+#ifdef DPRINTS
+  std::uint64_t srcHost = graph.getPhysicalHostID(src);
+  std::cout << "Source is on host " << srcHost << std::endl;
```
Comment (AdityaAtulTewari) [inline microbench/bfs/include/pando-bfs-galois/sssp.hpp:304]: `std::cerr`

File: microbench/bfs/include/pando-bfs-galois/sssp.hpp (hunk 1 above)
Comment (AdityaAtulTewari) [reply microbench/bfs/include/pando-bfs-galois/sssp.hpp:295]: oh i see no you are right.

**Comment by AdityaAtulTewari** [comment]
Fix the prints to be in cerr. and this looks good.

**Review by AdityaAtulTewari — APPROVED** [review]

**Pull request merged**